os/board/rtl8730e/src/component/mbed/targets/hal/rtl8730e/i2c_api.c

```diff
@@ -273,23 +273,30 @@ int rtk_i2c_read(i2c_t *obj, int address, char *data, int length, int stop)
 {
 	/* To avoid gcc warnings */
 	(void) stop;
-
+	int data_read = 0;
+	int cnt = 10;
 	if (i2c_target_addr[obj->i2c_idx] != address) {
 		i2c_prepare(obj, address);
 	}
 
 	if (!master_addr_retry) {
 		I2C_MasterRead(obj->I2Cx, (unsigned char *)data, length);
 	} else {
-		while (0 == I2C_MasterRead(obj->I2Cx, (unsigned char *)data, length)) {
+		while ((data_read = I2C_MasterRead(obj->I2Cx, (unsigned char *)data, length)) == 0) {
```

Comment:
Is there any chance to be less size than the length?
Regardless of the length, if i2c gives some data, is it ok to return without gathering all datas?

Reply:
Oh.. I saw this partially. (I did not check the change in ameba_i2c.c). It's ok.